oceanspy/tests/test_subsample.py

```diff
@@ -460,45 +492,178 @@ def test_survey(od, cartesian, delta, kwargs):
 lons170W = -170 * np.ones(np.shape(lats_6E))
 
 
-@pytest.mark.parametrize("this_od", [ECCOod])
+# ==
+y0 = [k for k in range(45, 80)]
+x0 = len(y0) * [40]
+
+# face 1
+y11 = [k for k in range(0, 11)]
+x11 = [49 + 4 * k for k in range(11)]
+
+y12 = [k for k in range(11, 16)]
+x12 = len(y12) * [89]
+
+x13 = [k for k in range(45, 60)]
+y13 = len(x13) * [25]
+
+y14 = [k for k in range(25, 35)]
+x14 = len(y14) * [89]
+
+x15 = [k for k in range(45, 60)]
+y15 = len(x15) * [45]
+
+x16 = [k for k in range(70, 80)]
+y16 = len(x16) * [89]
+
+x17 = [k for k in range(80, 85)]
+y17 = len(x17) * [79]
+
+x1 = x11 + x12 + x13 + x14 + x15 + x16 + x17
+y1 = y11 + y12 + y13 + y14 + y15 + y16 + y17
+
+# face 4
+x41 = [k for k in range(5, 46)]
+y41 = len(x41) * [60]
+y42 = [k for k in range(59, 10, -1)]
+x42 = len(y42) * [45]
+x4 = x41 + x42
+y4 = y41 + y42
+
+# face 3
+y31 = [k for k in range(80, 40, -1)]
+x31 = len(y31) * [60]
+x32 = [k for k in range(60, 5, -1)]
+y32 = len(x32) * [40]
+x3 = x31 + x32
+y3 = y31 + y32
+
+# face 0 again!
+x01 = [k for k in range(80, 45, -1)]
+y01 = len(x01) * [45]
+
+# cyclonic orientation
+Xc, Yc = [x0, x1, x4, x3, x01], [y0, y1, y4, y3, y01]
+
+# faces
+faces1 = [1, 2, 5, 4, 1]
+
+# extract lats/lons from data
+lonsc, latsc = [], []
+for i in range(len(Xc)):
+    for j in range(len(Xc[i])):
+        lonsc.append(
+            ECCOod._ds["XC"]
+            .isel(face=faces1[i], X=Xc[i][j], Y=Yc[i][j])
+            .squeeze()
+            .values
+        )
+        latsc.append(
+            ECCOod._ds["YC"]
+            .isel(face=faces1[i], X=Xc[i][j], Y=Yc[i][j])
+            .squeeze()
+            .values
+        )
+
+lons1, lats1 = [], []
+for j in range(len(Xc[1])):
+    lons1.append(
+        ECCOod._ds["XC"].isel(face=faces1[1], X=Xc[1][j], Y=Yc[1][j]).squeeze().values
+    )
+    lats1.append(
+        ECCOod._ds["YC"].isel(face=faces1[1], X=Xc[1][j], Y=Yc[1][j]).squeeze().values
+    )
+
+
+lons0, lats0 = [], []
+for j in range(len(Xc[0])):
+    lons0.append(
+        ECCOod._ds["XC"].isel(face=faces1[0], X=Xc[0][j], Y=Yc[0][j]).squeeze().values
+    )
+    lats0.append(
+        ECCOod._ds["YC"].isel(face=faces1[0], X=Xc[0][j], Y=Yc[0][j]).squeeze().values
+    )
+
```

Comment:
This is probably the section you are testing. Is this used multiple times? If so, it might be cleaner to include it in a pytest.fixture. This is not necessary of course.

Reply:
Yes, definitely good catch. I think at some point I kept redefining variables near the tests where they were used to keep track of all the testing done here. But you're right about pytest.fixture. Thanks!